org/postgresql/test/jdbc2/JdbcTimestampTest.java

```diff
@@ -0,0 +1,214 @@
+/*-------------------------------------------------------------------------
+*
+* Copyright (c) 2004-2014, PostgreSQL Global Development Group
+*
+*
+*-------------------------------------------------------------------------
+*/
+package org.postgresql.test.jdbc2;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Properties;
+import java.util.TimeZone;
+
+import junit.framework.TestCase;
+
+import org.postgresql.PGProperty;
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.PGTimestamp;
+
+/**
+ * Tests {@link java.sql.Timestamp and PGTimestamp} in various scenarios including setTimestamp, setObject for both
+ * <code>timestamp with time zone</code> and <code>timestamp without time zone</code> data types.
+ */
+public class JdbcTimestampTest extends TestCase
+{
+    /** The name of the test table. */
+    private static final String TEST_TABLE = "testjdbctimestamp";
+
+    /** The database connection. */
+    private Connection con;
+
+    public JdbcTimestampTest(final String name)
+    {
+        super(name);
+    }
+
+    protected void setUp() throws Exception
+    {
+        Properties props = new Properties();
+        props.setProperty(PGProperty.STRICT_TIMESTAMP.getName(), "true");
+        con = TestUtil.openDB(props);
+        TestUtil.createTable(con, TEST_TABLE, "ts timestamp, tz timestamp with time zone");
+    }
+
+    protected void tearDown() throws Exception
+    {
+        TestUtil.dropTable(con, TEST_TABLE);
+        TestUtil.closeDB(con);
+    }
+
+    public void testParameterIsNull() throws SQLException{
+        long now = System.currentTimeMillis();
```

Comment:
Please make tests reproducible.

In other words: use predefined hand-crafted timestamps.
If you want random tests, that is fine, however then pease use separate test and print random seed (see here: https://github.com/Gordiychuk/pgjdbc/commit/b1446130c73a9cdc31f8375da4091f2423fa986b#diff-41befd72881e4a629df6fc96be01098eR20)